lib/chunks.sql

```diff
@@ -190,27 +190,83 @@ WITH
         AND a.course_instance_id = ci.id
       )
   ),
-  chunks_to_delete AS (
+  -- We use separate CTEs for each chunk type so that Postgres doesn't tye to
+  -- compute a huge cross product of the chunks + metadata on `type` as an
+  -- intermediate step. This was O(n^2) with the number of chunks to delete.
```

Comment:
Rather than having separate CTEs and then an extra one to UNION them all together, we could do the UNIONs directly within a single top-level CTE like this: https://github.com/PrairieLearn/PrairieLearn/blob/79081d269ead89e8a013399c0d72119ac62ad295/lib/assessment.sql#L171

This would give us something like:
```
chunks_to_delete AS (
  (
    -- question chunks
    SELECT id
    FROM ...
    WHERE chunks.type = 'question'
  )
  UNION
  (
    -- client_files_course_instance chunks
    SELECT id
    FROM ...
    WHERE chunks.type = 'clientFilesCourseInstance'
  )
  ... etc
)
```

I'm fine with the current approach if you prefer it, however.

Reply:
Done in 996c887.